### ISSUE TITLE
Error message not very useful

### ISSUE THREAD

**Issue by nikwest** [IBM-Swift/BluePic#376]
Just trying out BluePic. When starting locally I get:

> ./BluePicServer
 INFO: init() Configuration.swift line 51 - Using configuration values from 'cloud_config.json'. 
 ERROR: BluePicServer main.swift line 34 - Oops, something went wrong... Server did not start! Failed with error: The operation couldn’t be completed. (BluePicApp.Configuration.BluePicError error 0.) 

which is not very helpful:

```swift
diff --git i/BluePic-Server/Sources/BluePicServer/main.swift w/BluePic-Server/Sources/BluePicServer/main.swift
index 1d3affa..0d0e256 100644
--- i/BluePic-Server/Sources/BluePicServer/main.swift
+++ w/BluePic-Server/Sources/BluePicServer/main.swift
@@ -24,13 +24,13 @@ import CloudFoundryDeploymentTracker
 HeliumLogger.use(LoggerMessageType.info)
 
 do {
   let serverController = try ServerController()
   // Start server...
   Kitura.addHTTPServer(onPort: serverController.port, with: serverController.router)
   Kitura.run()
 
-} catch let error {
-  Log.error("Oops, something went wrong... Server did not start! Failed with error: \(error.localizedDescription)")
-  exit(1)
+} catch Configuration.BluePicError.IO(let reason) {
+    Log.error("Oops, something went wrong... Server did not start! Failed with error: \(reason)")
+    exit(1)
 }
```

will fix it and return:

> BluePicServer 
 INFO: init() Configuration.swift line 51 - Using configuration values from 'cloud_config.json'. 
 ERROR: BluePicServer main.swift line 34 - Oops, something went wrong... Server did not start! Failed with error: Failed to obtain IBM Push credentials.

**Comment by tfrank64** [comment]
@nikwest You are correct, it would be much more helpful to print out descriptive errors. We would love to see your proposed fix on this. You could also look into LocalizedErrors, similar to what we do in [RouteUtils](https://github.com/IBM-Swift/BluePic/blob/master/BluePic-Server/Sources/BluePicApp/RouteUtils.swift#L27). To actually accept a pull request, we need you to have completed a [CLA form](https://github.com/IBM-Swift/CLA). Thanks

**Comment by quanvo87** [comment]
I fixed the issue by replacing "imfpush"s entry in cloud_config.json with the whole "imfpush" you get when you pull up the service credentials for BluePic-IBM-Push in your app's page on Bluemix.

For example, mine is:
```
"imfpush": [
    {
      "credentials": {
        "appGuid": "1d47e593-dcc3-4d7c-bd64-3669a0ec7255",
        "url": "http://imfpush.ng.bluemix.net/imfpush/v1/apps/1d47e593-dcc3-4d7c-bd64-3669a0ec7255",
        "admin_url": "//mobile.ng.bluemix.net/imfpushdashboard/?appGuid=1d47e593-dcc3-4d7c-bd64-3669a0ec7255",
        "appSecret": "362697d1-9245-4160-a451-5302e484084f",
        "clientSecret": "13cecaff-3a24-491d-94f8-ff440007537f"
      },
      "syslog_drain_url": null,
      "label": "imfpush",
      "provider": null,
      "plan": "Basic",
      "name": "BluePic-IBM-Push",
      "tags": [
        "mobile",
        "ibm_created",
        "ibm_dedicated_public"
      ]
    }
  ]
```

**Comment by YouYue123** [comment]
Hi,

I found the same issue when I am running in my local docker container.

The reason is because in cloud_config file for imfpush configuration. It is missing  appGuid field which is required in Configuration.swift as below.

```
    guard let appGuid = pushCredentials["appGuid"].string,
      let url = pushCredentials["url"].string,
      let adminUrl = pushCredentials["admin_url"].string,
      let secret = pushCredentials["appSecret"].string else {
        throw BluePicError.IO("Failed to obtain IBM Push credentials.")
    }
    let ibmPushProperties = IbmPushProps(appGuid: appGuid, url: url, adminUrl: adminUrl, secret: secret)
```

And from the code above, I can see it has already tried to throw a useful message. But it is not shown in the logging. Because in main.swift, the error is in the following format.

`IO("Failed to obtain IBM Push credentials.")`
`

But the localizedDescription attribution for this error will always show the following message:

```
The operation couldn’t be complete
d. (BluePicApp.Configuration.BluePicError error 0.)
```

So I removed localizedDescrtion show the useful message as below:

`Log.error("Oops, something went wrong... Server did not start! Failed with error: \(error)")`

Now it is showing the correct log as below:

`ERROR: BluePicServer main.swift line 35 - Oops, something went wrong... Server did not start! Failed with error: IO("Failed to obtain IBM Push credentials.") `

Do you think this is acceptable? If yes, I am willing to submit a pull request.

**Comment by tfrank64** [comment]
@YouYue123 Good catch with the `cloud_config.json` `appGuid` missing field. We will need to update that file.

As for the error logging issue, that looks like an acceptable fix. Once you have submitted a [CLA](https://github.com/IBM-Swift/CLA) form, please feel free to submit a pull request for this!

**Comment by tfrank64** [comment]
FYI, I went ahead and updated our cloud_config.json in a [PR](https://github.com/IBM-Swift/BluePic/pull/383).

**Comment by tfrank64** [comment]
Looks like @christiancompton fixed this in the latest PR merge #387
